Clarification of narrative, adding brief mentions to automated unit testing, and providing response to reviewer comments (#307)
SORTEE-Github-Hackathon/manuscript: content/03.use-cases-a.md

An EEB research compendium includes all computational materials related to research production, including data, code for analyses and protocols. 
Safely storing these files is essential to protect against accidental modifications or deletions.
Many researchers begin using GitHub to backup their research compendium [@doi:10.1080/00031305.2017.1375986] into a centralized, readily-available remote server (see [Box 1](#definitions)).
This practice has the advantages of facilitating collaboration, integrating data and code archiving, allowing file versions to be accessed and restored, and further contributing to open science practices (@doi:10.1162/99608f92.9497f68e).
Every change made to files in version-controlled repositories accompany authored descriptions of modifications ([Box 1](#definitions)).

> Later, the entire history of commits and their commit messages are viewable and can be audited, similarly as with physical laboratory notebooks [@doi:10.1186/1751-0473-8-7].

Reviewer: ```suggestion
Later, the entire history of commits and their commit messages are viewable and can be audited similarly to physical laboratory notebooks [@doi:10.1186/1751-0473-8-7].
```

> GitHub allows administrative control over who can make modify repositories, and the process of cloning, forking and suggesting changes through pull requests (see [Box 1](#definitions)) ensures code owners have full control over code and documents.

Reviewer: ```suggestion
GitHub allows administrative control over who can view and modify repositories, and the process of cloning, forking and suggesting changes through pull requests (see [Box 1](#definitions)) ensures code owners have full control over code and documents.
```

> Repositories can be changed from public to private, allowing data storage and management without sacrificing the privacy that may be necessary in certain compendia.

Reviewer: ```suggestion
Repositories can be made public or private (and changed at any time), allowing data storage and management without sacrificing the privacy that may be necessary in certain compendia.
```

However, users can still track versions of remotely- or locally-stored files in their repositories using Git Large File Storage (<https://git-lfs.github.com>).

### Project continuity

<!--*Contributors to this section: BPME, VF, PHPB  -->

Projects in ecology and evolution often involve research professionals holding limited-term positions, such as graduate students, research assistants, and post-doctoral fellows [@doi:10.5334/jors.307].
Without clear plans on project continuity, research code and data management upkeep tends to fall off as researchers move on to new projects or other institutions.
Additionally, code and data can be difficult to access and recover when kept only on personal devices [@doi:10.1016/j.cub.2013.11.014].

GitHub can facilitate project continuity in research by making code and data handover between users easier [@doi:10.5334/jors.307; @doi:10.1186/1751-0473-8-7].
Through version control, the history of code and data from projects in ecology and evolution becomes accessible to future laboratory members and collaborators [@doi:10.1038/s41559-017-0160].

> Repositories and organizations can have designated data and code owners [or more appropriate, "data stewards"; see @doi:10.1890/ES14-00402.1, @url:https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners], who can also change through time, allowing for the transition of code between research cohorts (see also ["Organizing and managing teams"](#organizations)).

Reviewer: ```suggestion
Repositories and organizations can have designated data and code owners [or more appropriately, "data stewards"; see @doi:10.1890/ES14-00402.1, @url:https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners], who can also change through time, allowing for the transition of code between research cohorts (see also ["Organizing and managing teams"](#organizations)).
```

Other project collaborators can contribute to repository design and development, and their active involvement can both aid authors ability to act as guarantors, and the clarity and reproducibility of the project for future users.
In (Figure @fig:github-diagram), we highlight several elements of recommended repository structure, and the various ways that contributors may interact with them.

Software compatibility during the analysis and reanalysis of project data can be ensured by storing information about software dependencies and their versions within the same project repository.

> With more advanced practices, one can remotely install and execute scripts using specific versions of software within GitHub's project automation tools, GitHub Actions (see below).

Reviewer 2: This sentence and the one above might be too technical for this manuscript? I'm not sure. We're talking indirectly about things like `conda`, `pipenv` and `requirements.txt` files. Would it be helpful to call them out directly or just remove these two sentences because this type of work falls under a more advanced use case.

Reviewer: I agree—if we are talking about tracking versions of dependencies, then that is not really a GitHub-specific thing (although it is very useful for getting GitHub actions working)

### Educational materials

<!-- *contributors to this section: Cole Brookson, PHPB* -->

GitHub supports a broad set of mechanisms for hosting educational materials.
The entire process of running a course, workshop, or lecture, can all be done openly on GitHub including material development, web hosting, and delivery, and even submission and grading of assignments.
While there are other purpose-built platforms for this, GitHub provides a free, open-source alternative.

Syllabi, presentations, and other course materials can be version-controlled, stored and managed in GitHub.

> While any file type for presentations and documents can be stored and tracked in repositories, GitHub can directly render and display certain markup languages, such as HyperText Markup Language (HTML) and Markdown documents, making these plain-text documents especially advantageous for educational content files.

Reviewer 2: ```suggestion
While any file type for presentations and documents can be stored and tracked in repositories, GitHub can directly render and display certain markup languages, such as HyperText Markup Language (HTML) and Markdown documents, making these file types especially useful for educational materials.
```

> Course content can then be delivered to the audience by sharing links of web-hosted Github Pages (see [Hosting a Website](#hosting-a-website)), GitHub Repositories and/or GitHub Organizations.

Reviewer 2: ```suggestion
Course content can then be delivered to the audience by sharing links of web-hosted GitHub Pages (see [Hosting a Website](#hosting-a-website)), GitHub Repositories and/or GitHub Organizations.
```
